erpnext/payroll/doctype/overtime_slip/overtime_slip.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _, bold
+from frappe.utils import get_datetime, getdate, get_link_to_form, formatdate
+from erpnext.payroll.doctype.payroll_entry.payroll_entry import get_start_end_dates
+from erpnext.payroll.doctype.gratuity.gratuity import get_salary_structure
+from frappe.model.document import Document
+class OvertimeSlip(Document):
+	def validate(self):
+		if not (self.from_date or self.to_date or self.payroll_frequency):
+			date = self.from_date or self.posting_date
+			data = get_frequency_and_dates(self.employee, date)
```

Comment:
We can call this function `set_payroll_details`, directly set the fields in this method and move it to the class instead. We can call it from client-side using `frm.call`